app/middleware/authentication.rb

```diff
@@ -41,6 +34,15 @@ def call(env)
 
     private
 
+    def update_user_email_if_needed(user, email)
+      return unless user.present? && email.present?
```

Comment:
If we move this method down a couple lines so it's inside the conditional on line 25, we can drop the `user.present?`.
Also, we know that the email exists by this point, or there was an "email" key in the Id token, anyway. If this value is nil or an empty string, then we won't be able to save this record. It will already have raised a validation error if the user is new, and later we'll be using `save!` so we won't be setting the email to an invalid value if we're updating an existing user.